### PR TITLE
Add ResourceQuota and LimitRange to Org Namespace

### PR DESCRIPTION
**Pull request by glrf**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -33,3 +33,32 @@ parameters:
     generatedDefaultRoleBindingInNewNamespaces:
       bindingName: admin
       clusterRoleName: admin
+
+    generatedResourceQuota:
+      name: organization
+      synchronize: false
```
Comment (ccremer) [inline class/defaults.yml:39]: To my understanding, we shall prevent modifications altogether.
Consider the case where we want to adjust the quota for all namespaces, it should be enough to modify the cluster policy that takes care to synchronize all the existing quotas.
```suggestion
      synchronize: true
```
Yes, at some point we want to make them adjustable for special namespaces, but that's for later.

File: class/defaults.yml [hunk 2]
```diff
@@ -33,3 +33,32 @@ parameters:
...
+      hard:
+        requests.cpu: "1"
+        requests.memory: 4Gi
+        limits.cpu: "2"
+        limits.memory: 8Gi
```
Comment (ccremer) [inline class/defaults.yml:44]: We have some values defined in our requirements: https://kb.vshn.ch/appuio-cloud/references/quality-requirements/performance/resource-quota.html
Whether these values are sensible is another question, but please keep them in sync at least.
```suggestion
        # See https://kb.vshn.ch/appuio-cloud/references/quality-requirements/performance/resource-quota.html
        requests.cpu: "1"
        requests.memory: 4Gi
        limits.cpu: "2"
        limits.memory: 8Gi
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -91,3 +91,156 @@ default:: `admin`
...
+    min:
+      cpu: "10m"
+      memory: "4Mi"
+    default:
+      cpu: "200m"
+      memory: "200Mi"
+    defaultRequest:
+      cpu: "100m"
+      memory: "100Mi"
+----
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:222]: I don't think it makes sense to copy-paste the entire block here. I would just refer to the `defaults.yaml`

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 4]
```diff
@@ -91,3 +91,156 @@ default:: `admin`
...
+  "Container":
+    max:
+      memory: "4G"
+    min:
+      memory: "4Mi"
+  "PersistentVolumeClaim":
+    max:
+      storage: 2Gi
+    min:
+      storage: 1Gi
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:238]: Same here

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 5]
```diff
@@ -91,3 +91,156 @@ default:: `admin`
...
+default::
++
+[source,yaml]
+----
+hard:
+  requests.cpu: "1"
+  requests.memory: 4Gi
+  limits.cpu: "2"
+  limits.memory: 8Gi
+----
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:132]: and here.

**Review by ccremer — CHANGES_REQUESTED** [review]

File: class/defaults.yml (hunk 2 above)
Comment (glrf) [reply class/defaults.yml:44]: As "choose reasonable defaults" is explicitly out of scope, I just chose something arbitrary.  But sure I can switch to these for now.

File: docs/modules/ROOT/pages/references/parameters.adoc (hunk 4 above)
Comment (glrf) [reply docs/modules/ROOT/pages/references/parameters.adoc:238]: That's an example. And especially as right now we only have a default for `Container`, I would leave that in.

**Review by corvus-ch — APPROVED** [review]
New insights on the subject of ResouceQuota requires to have the ability to configure more than one object. It quick to change, please do it. However I am also fine with having it done in APUB-48 instead.

.

File: class/defaults.yml [hunk 6]
```diff
@@ -33,3 +33,40 @@ parameters:
...
+        persistentvolumeclaims: "50"
+        count/configmaps: "150"
+        count/secrets: "150"
+        count/services: "20"
+        openshift.io/imagestreams: "20"
+        openshift.io/imagestreamtags: "50"
+      scopes: []
+      scopeSelector:
+
+    generatedLimitRange:
```
[the remark this answers is not in the document]
Comment (glrf) [reply class/defaults.yml:56]: I don't think it is. I think you should be able to express any possible limits with a single LimitRange

**Review by ccremer — APPROVED** [review]

File: class/defaults.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ccremer) [reply class/defaults.yml:56]: Though I'd go for consistency here, let's go with the current approach. Might still change later (even if breaking)

**Pull request merged**